android/app/src/main/java/com/expensify/chat/bootsplash/BootSplashModule.java

```diff
@@ -47,6 +48,19 @@ public String getName() {
     return NAME;
   }
 
+  // From https://stackoverflow.com/a/61062773
+  public static boolean isSamsungOneUI4() {
+    String name = "SEM_PLATFORM_INT";
+
+    try {
+      Field field = Build.VERSION.class.getDeclaredField(name);
+      int version = (field.getInt(null) - 90000) / 10000;
+      return version == 4;
+    } catch (Exception ignored) {
+      return false;
+    }
+  }
```

Comment:
I will apply the same in `react-native-bootsplash` in a upcoming version.

This is not pretty, but it's extracted from a Samsung APK so it should be correct, and we unfortunately need it 😕

I have a similar method in `react-native-localize` for Xiaomi devices. I always rather use a workaround than let users reporting issues, as they don't know that it's their phone Android implementation fault.

I often wish Google enforce Android stock usage (without any modifications) to manufacturers, but it's a wild dream 😅

Reply:
> I will apply the same in react-native-bootsplash in a upcoming version.

Okay, that's great. Can you let us know when this is ready so that we can update our code?

>I often wish Google enforce Android stock usage 

As an ex-native Android dev I 100% agree.

Reply:
@Julesssss Yes, I will. The module in Expensify is based on my initial work on v5, but has been merged way before I released this one (as I added support for brand image, dark mode, web, etc)

Once bootsplash catch up on this, I will do a proposal to migrate Expensify back to it (and delete the custom module in your codebase). I know that light / dark mode support is planned on your side + you will be able to remove your web custom implementation.

Meanwhile, I'd rather be able to merge and check that this indeed fix the Samsung issue (as I don't have such test devices)